docs/src/man/reshaping_new.md

```diff
@@ -0,0 +1,337 @@
+──────┼─────────────────────────────────────────────────────────────────────────
+    1 │     0     67  male        2  own      missing          little          ⋯
+    2 │     1     22  female      2  own      little           moderate
+    3 │     2     49  male        1  own      little           missing
+    4 │     3     45  male        2  free     little           little
+    5 │     4     53  male        2  free     little           little          ⋯
+    6 │     5     35  male        1  free     missing          missing
+    7 │     6     53  male        2  own      quite rich       missing
+    8 │     7     35  male        3  rent     little           moderate
+  ⋮   │   ⋮      ⋮      ⋮       ⋮       ⋮            ⋮                ⋮        ⋱
+  994 │   993     30  male        3  own      little           little          ⋯
+  995 │   994     50  male        2  own      missing          missing
+  996 │   995     31  female      1  own      little           missing
+  997 │   996     40  male        3  own      little           little
+  998 │   997     38  male        2  own      little           missing         ⋯
+  999 │   998     23  male        2  free     little           little
+ 1000 │   999     27  male        2  own      moderate         moderate
+                                                  4 columns and 985 rows omitted
+
+julia> german = view(german_ref, 1:10, 1:6)
+10×6 SubDataFrame
+ Row │ id     Age    Sex     Job    Housing  Saving accounts
+     │ Int64  Int64  String  Int64  String   String?
+─────┼───────────────────────────────────────────────────────
+   1 │     0     67  male        2  own      missing
+   2 │     1     22  female      2  own      little
+   3 │     2     49  male        1  own      little
+   4 │     3     45  male        2  free     little
+   5 │     4     53  male        2  free     little
+   6 │     5     35  male        1  free     missing
+   7 │     6     53  male        2  own      quite rich
+   8 │     7     35  male        3  rent     little
+   9 │     8     61  male        1  own      rich
+  10 │     9     28  male        3  own      little                                                  
+```
+
+## The basics of `stack`
+
+Stack return the long-format `DataFrame` with columns for each of the ID variables, column 
+value name holding the values of the stacked columns, and column variable name a vector holding 
+the name of the corresponding measurement variables.
+
+Here, `german` is the `AbstractDataFrame` which has to be stacked and the columns to
+be stacked are `:Age`, `:Job`, `:Housing`, `:Sex` which is known as the measurement variables;
+`:variable` is the name of our new stacked columns which are holding the names of each of the
+measurement variables, and `value` is the name of the new stacked columns containing the values
+from each of the measurement variables:
+
+```jldoctest dataframe
+julia> stack(german, [:Age, :Job, :Housing])
```

Comment:
I would recommend you to start with examples showing all arguments passed explicitly